data_types/softlayer_dns_domain.go

```diff
@@ -0,0 +1,22 @@
+package data_types
+
+type SoftLayer_Dns_Domain_Template struct {
+	Name            string                        `json:"name"`
+	ResourceRecords []SoftLayer_Dns_Domain_Record `json:"resourceRecords"`
+}
+
+type SoftLayer_Dns_Domain_Template_Parameters struct {
+	Parameters []SoftLayer_Dns_Domain_Template `json:"parameters"`
+}
+
+type SoftLayer_Dns_Domain struct {
+	Id         int    `json:"id"`
+	Name       string `json:"name"`
+	Serial     int    `json:"serial"`
+	UpdateDate string `json:"updateDate"`
+	//	Account					SoftLayer_Account 		`json:"account"`
```

Comment:
why are these commented out, you can use `omitempty` to make them optional... Same for the others. See in the other `data_types` how we did that.